Trouble finding includes for OpenSSL on OS X 10.10/10.11

On OS X 10.10 & 10.11, System.Security.Cryptograhpy.Native is having problems finding the copy of OpenSSL found by homebrew on my system. I originally get the error:

```
/corefx/src/Native/System.Security.Cryptography.Native/openssl.c:14:10: fatal error: 'openssl/asn1.h' file not found
#include <openssl/asn1.h>
```

If I work around this issue by adding OPENSSL_INCLUDE_DIR to the target header search path, I then have errors with warnings in the headers. For example:

```
/usr/local/include/openssl/ecdsa.h:295:14: error: parameter 'flags' not found in
      the function declaration [-Werror,-Wdocumentation]
/usr/local/include/openssl/objects.h:1025:10: error: macro name is a reserved
      identifier [-Werror,-Wreserved-id-macro]

/Sources/dotnet/corefx/src/Native/System.Security.Cryptography.Native/pal_err.cpp:25:27: error: 
      use of old-style cast [-Werror,-Wold-style-cast]
        *isAllocFailure = ERR_GET_REASON(err) == ERR_R_MALLOC_FAILURE;
                          ^~~~~~~~~~~~~~~~~~~
/usr/local/include/openssl/err.h:246:34: note: expanded from macro
      'ERR_GET_REASON'
# define ERR_GET_REASON(l)       (int)((l)&0xfffL)
```

Is this user error or a problem with the build? If it's a problem with the build, any suggestions on what kind of fix is most likely to be accepted back?

Tested on OS X 10.11
Apple LLVM version 7.3.0 (clang-703.0.29)
Target: x86_64-apple-darwin15.4.0
Thread model: posix

Homebrew 0.9.9 (git revision ac80; last commit 2016-04-03)
Homebrew/homebrew-core (git revision 0db2; last commit 2016-04-03)

openssl: stable 1.0.2g

I'm seeing the same issue on OSX 10.11.4 that openssl/*.h is not being found - I have installed 1.0.2f openssl through homebrew and did the link. After adding OPENSSL_INCLUDE_DIR to CMakeList.txt, I'm now seeing slightly different build failures: 

usr/local/include/openssl/stack.h:68:12: error: padding struct
      'struct stack_st' with 4 bytes to align 'data' [-Werror,-Wpadded]
    char **data;
           ^
[ 20%] Building CXX object System.Net.Http.Native/CMakeFiles/System.Net.Http.Native.dir/pal_versioninfo.cpp.o
In file included from /Users/yizhang/github/corefx/src/Native/System.Security.Cryptography.Native/openssl.c:14:
In file included from /usr/local/include/openssl/asn1.h:65:
/usr/local/include/openssl/bio.h:790:11: error: macro name is a reserved
      identifier [-Werror,-Wreserved-id-macro]
#  define __bio_h__attr__ __attribute__

I can submit a PR for the include issue, but I suspect openssl is not compatible with XCode 7.3 that I have.

--- a/src/Native/configure.cmake
+++ b/src/Native/configure.cmake
@@ -11,6 +11,10 @@ if (CMAKE_SYSTEM_NAME STREQUAL Linux)
     set(PAL_UNIX_NAME \"LINUX\")
 elseif (CMAKE_SYSTEM_NAME STREQUAL Darwin)
     set(PAL_UNIX_NAME \"OSX\")
+
+    # Xcode's clang does not include /usr/local/include by default, but brew's does.
+    # This ensures an even playing field.
+    include_directories(SYSTEM /usr/local/include)
 elseif (CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
     set(PAL_UNIX_NAME \"FREEBSD\")
     include_directories(SYSTEM /usr/local/include)

